clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -692,6 +692,8 @@ protected void onJoinPrepare(int generation, String memberId) {
 
     @Override
     public void onLeavePrepare() {
+        log.debug("Executing onLeavePrepare with generation {} and memberId {}", generation(), memberId());
```

Comment:
Since generation may change by the hb in between this line and line 702 below, this entry may not have the exact number, maybe just read out from `generation()` and use that object?

Reply:
+1

Reply:
You're referring to `memberId()` right? It seems like `generation().memberId` returns the same thing as `memberId()` -- or do you mean I should first check `generation().hasMemberId()` and then, if that is empty, do what -- just log nothing, or a separate message for when memberId is empty?

Reply:
No I'm referring to the `generation()` itself: we call this function twice within the function and in between the generation object may have been changed.

Reply:
Ohh right ok, I'll fix that